docs/features/techdocs/configuration.md

```diff
@@ -0,0 +1,72 @@
+  Reference documentation for configuring TechDocs using app-config.yaml
+---
+
+Using the `app-config.yaml` in the Backstage app, you can configure TechDocs
+using several options. This page serves as a reference to all the available
+configuration options for TechDocs.
+
+```yaml
+# File: app-config.yaml
+
+techdocs:
+
+  # TechDocs makes API calls to techdocs-backend using this URL. e.g. get docs of an entity, get metadata, etc.
+
+  requestUrl: http://localhost:7000/api/techdocs
+
+
+  # Just another route in techdocs-backend where TechDocs requests the static files from. This URL uses an HTTP middleware
+  # to serve files from either a local directory or an External storage provider.
+
+  storageUrl: http://localhost:7000/api/techdocs/static/docs
+
+
+  # generators.techdocs can have two values: 'docker' or 'local'. This is to determine how to run the generator - whether to
+  # spin up the techdocs-container docker image or to run mkdocs locally (assuming all the dependencies are taken care of).
+  # You want to change this to 'local' if you are running Backstage using your own customer Docker setup and want to avoid running
+  # into Docker in Docker situation. Read more here
+  # https://backstage.io/docs/features/techdocs/getting-started#disable-docker-in-docker-situation-optional
+
+  generators:
+    techdocs: 'docker'
+
+
+  # techdocs.builder can be either 'local' or 'ci.
+  # If builder is set to 'local' and you open a TechDocs page, techdocs-backend will try to build the docs, publish to storage
+  # and show the generated docs afterwords. This is the "Basic" setup of the TechDocs Architecture.
+  # If builder is set to 'ci', techdocs-backend will only fetch the docs and will NOT try to build and publish. In this case of 'ci',
+  # we assume that docs are being built in the CI/CD pipeline of the repository. This is the "Recommended" setup of the architecture.
+  # Read more here https://backstage.io/docs/features/techdocs/architecture
+
+  builder: 'local'
+
+
+  # techdocs.publisher is used to configure the Storage option, whether you want to use the local filesystem to store generated docs
+  # or you want to use External storage providers like Google Cloud Storage, AWS S3, etc.
+
+  publisher:
+
+    # techdocs.publisher.type can be - 'local' or 'google_gcs' (aws_s3, azure_storage, etc. to be available as well).
+    # When set to 'local', techdocs-backend will create a 'static' directory at its root to store generated documentation files.
+    # When set to 'google_gcs', techdocs-backend will use a Google Cloud Storage Bucket to store generated documentation files.
+
+    type: 'local'
+
+
+    # Required when techdocs.publisher.type is set to 'google_gcs'. Skip otherwise.
+
+    google:
+      # An API key is required to write to a storage bucket.
+      pathToKey: '/path/to/google_application_credentials.json',
+
+      # Your GCP Project ID where the Cloud Storage Bucket is hosted.
+      projectId: 'gcp-project-id'
```

Comment:
Oh we need this? Pity, I hoped we wouldn't have to, given that bucket names are global

Reply:
@freben Technically we don't need this to make the process work. However, I thought it might help the developer easily locate the following bucket if they know the name of its GCP project.

I could make this an optional config, I believe.

Reply:
Leaning towards leaving projectId in the config. The client does make some API calls which uses `projectId`. Plus it seems to be the recommended way of defining the client
https://github.com/googleapis/nodejs-storage/blob/5fa530ab909e837558deecd7e1aa50a7cc4f5830/src/storage.ts#L385-L390